docs/validators/node-upgrades.md

```diff
@@ -16,27 +16,29 @@ Currently, we have the following types of upgrades:
 A. - **all nodes need to upgrade**:  upgrades that involve processing changes with an activation epoch (as explained below) 
 and have to be performed by all nodes operators in order to keep the same view over the network and not cause service disruptions.
 
-B. - **optional upgrades**: upgrades that for example simply add a new Rest API endpoint or improve the trie syncing timing 
+B. - **optional upgrades**: upgrades that, for example, simply add a new Rest API endpoint or improve the trie syncing timing 
 are not critical from a processing point of view, and they are optional. If the nodes operators think the new feature will help them,
 they can proceed with the upgrade without losing the compatibility with the network.
 
 C. - **only validators need to upgrade**: upgrades that, for example, include new features that only trigger validators (ratings changes, 
-transactions selection improvements and so on). Observers (nodes that don't have a stake attached) don't need to perform the upgrade.
+transactions selection improvements and so on). Observers (nodes that don't have a stake attached) don't need to perform the upgrade
+(but can upgrade nonetheless if desired).
 
 ## **Activation epochs**
 
-In order to make the upgrades as smooth as possible and to ensure that each node has the same view over the network, 
-Elrond has a so called *activation epoch* mechanism that allows the node to have both versions of the code (before and 
-after the activation epoch) so that we can maintain the backwards compatibility. The system will behave accordingly to 
-the epoch.
+In order to make the upgrades as smooth as possible and to ensure that each node has the same view over the network at a given moment, 
+Elrond has a so called *activation epoch* mechanism that allows the node to implement both behaviors of the protocol - 
+the old (current) one, and the new one, planned for activation at a specific epoch. This mechanisms ensures that, 
+**until the protocol change becomes active**, nodes with an upgraded codebase / binary remain compatible with nodes that 
```

Comment:
this is desirable but not always attainable
If we want to be politically correct we should state the following:
`This mechanisms ensures that, **until the protocol change becomes active**, nodes with an upgraded codebase / binary remain compatible with nodes that did not perform the upgrade, and consensus is held.
Although this happens in 99.9% cases, this is not 100% guaranteed due to the unforeseen consequences when upgrading the code base in parallel with executing transactions generated by 3rd parties (Elrond blockchain users)`

Reply:
added